src/Locations/saga.js

```
@@ -0,0 +1,55 @@
+import axios from 'axios';
+// import Toast from '../Shared/Toastify/Toast';
+import { all, put, select, takeEvery, takeLatest } from 'redux-saga/effects';
+
+function* fetchBranchLocations({ payload }) {
+  const authConfig = yield select(state => state.user.session);
+
+  try {
+    const res = yield axios.get(`/branches/${payload.branchId}`, authConfig);
+    console.log(res);
+    yield put({
+      type: 'FETCH_BRANCH_LOCATIONS_SUCCEEDED',
+      payload: {
+        branchId: res.data.id,
+        locations: res.data.branch_locations,
+      },
+    });
+  } catch (e) {
+    console.log(e);
+  }
+}
+
+function* updateBranchLocation({ payload }) {
+  const authConfig = yield select(state => state.user.session);
+  const value = yield select(
+    state =>
+      state.branches.data[payload.branchId].locations[payload.locationIndex]
+        .location[payload.field],
+  );
+
+  try {
+    yield axios.patch(
+      `/branches/${payload.branchId}/locations/${payload.locationId}`,
+      { [payload.field]: value },
+      authConfig,
+    );
+    yield put({
+      type: 'UPDATE_BRANCH_LOCATIONS_UPSTREAM_SUCCEEDED',
+    });
+  } catch (e) {
+    console.log(e);
```

Add Toast to indicate failure instead of logging it to console